### PR TITLE
Clean up the backend-defaults database report

### LINKED ISSUE #26357: Clean up backend-defaults/database
Updates affecting the api report:

- [x] `Config` -> `ConfigService`
- [ ] Remove  the `LegacyRootDatabaseService` type, make deps required on the regular class instead.
- [ ] Remove `logger` from `fromConfig` - no longer needed there
- [ ] Look into removing the `pluginMetadata` service dependency - we already pass in the plugin ID and it just needs to be passed down to the sqlite layer
- [ ] Remove `dropDatabase` which has been deprecated for a while

### PR DESCRIPTION
**Pull request by freben**
Closes #26357

See the changeset for the list of actual changes. The commits that make up the PR are fairly self contained and can be reviewed in isolation.

### CODE REVIEW

File: packages/backend-common/src/deprecated/index.ts [hunk 1]
```diff
@@ -186,7 +192,20 @@ export class DatabaseManager implements LegacyRootDatabaseService {
...
+      warn() {},
+      error() {},
+      child() {
+        return this;
+      },
+    };
+    const lifecycle: LifecycleService = deps?.lifecycle ?? {
+      addShutdownHook() {},
+      addStartupHook() {},
+    };
```
Comment (freben) [inline packages/backend-common/src/deprecated/index.ts:207]: Meh. We're gonna remove all of this soon anyway.

File: .changeset/wise-forks-play.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+'@backstage/backend-common': minor
```
Comment (Rugvip) [inline .changeset/wise-forks-play.md:2]: tbh I think we should avoid breaking changes in `backend-common`, just moving over the deprecated implementations to then be deleted after the next release

**Review by Rugvip — COMMENTED** [review]
Nice! 👍

File: .changeset/wise-forks-play.md (hunk 2 above)
Comment (Rugvip) [reply .changeset/wise-forks-play.md:2]: Alright, never mind let's ship this to avoid a messy refactor 😅

**Review by Rugvip — APPROVED** [review]

**Pull request merged**